server/api-service/lowcoder-server/src/test/java/org/lowcoder/api/authentication/AuthenticationControllerTest.java

```diff
@@ -84,7 +82,7 @@ public void testFormRegisterSuccess() {
                     assertEquals(email, connection.getRawId());
                     assertEquals(email, connection.getName());
                     assertNull(connection.getAvatar());
-                    assertEquals(1, connection.getOrgIds().size());
+                    assertEquals(0, connection.getOrgIds().size());
```

Comment:
Well, I debugged/walked the whole workflow/code path and realized that the org ids are set/added to the user only when the `AuthConfig` sent in the Login API call has an associated organization. In the case of the **Email** config the associated organization is null. Hence, the resulting organization ids are also empty.

For a more technical understanding:
1. `AuthConfigId` is sent to the login api call which contains the organization info (which is null in the case being discussed) via the `FormLopinRequest` record.
2. The `authenticate()` function in `AuthenticationApiServiceImpl` does the following:
	- Fetches the auth config for the given auth id
	- Set the auth config and org id into the context. If org is null ( which it is not in the case being discussed ), the org id in the context is also set to null.
	- Then build the reqFactory and do the actual auth
	- Then set the orgId from the context ( which is still null at this point) into the `AuthUser` object.
	- Return the auth user.
3. The `updateConnection()` function checks if the orgId is there in the authUser object and adds it to the orgIds List if available. Since, nothing is there, it never adds any org ids.

This is the current workflow. 

If you think that the workflow of setting orgs ids is wrong within the auth flow, then we might have a bug in the login/registration workflow.

Reply:
I understand this makes sense now for me.
Your change makes sense therefore.